src/main/java/org/opensearch/security/auth/limiting/AbstractRateLimiter.java

```diff
@@ -66,6 +77,16 @@ public WildcardMatcher getIgnoreHostsMatcher() {
         return hostMatcher;
     }
 
+    @Override
+    public SubnetUtils.SubnetInfo getSubnetForCidr(String cidr) {
+        return SUBNET_CACHE.computeIfAbsent(cidr, pattern -> {
```

Comment:
Why do we need a cache here? Can't the `SubnetUtils` instances be created upon initialization of an instance of this class (or a subclass)?

Reply:
The cache is meant to ensure dynamic changes are also captured.

Reply:
By dynamic changes, do you mean changes dynamic changes to the configuration (like `config.yml`, etc)?

The `AbstractRateLimiter` instances are stored in the list `ipClientBlockRegistry` in `BackendRegistry.java`. 

This list is updated upon changes of the dynamic configuration:

https://github.com/opensearch-project/security/blob/6efe00b6c1b978730dd9a0d6ef6e9d17ab2daec3/src/main/java/org/opensearch/security/auth/BackendRegistry.java#L178-L195

Thus, I'd believe it is possible to pre-compute the `SubnetUtils` instances during creation of `AbstractRateLimiter` instances and thus being able to avoid the use of any caches at all.

Reply:
By dynamic changes, I also mean changes made using Security APIs- https://opensearch.org/docs/latest/security/access-control/api/#patch-configuration

We could still pre-compute during initialization and later use a subscriber to update instances of SubnetUtils, thoughts?

Reply:
> By dynamic changes, I also mean changes made using Security APIs

I would believe that these also cause an invocation of onDynamicConfigModelChanged(), no?

Reply:
yes - made the changes to pre-compute.